packages/tenant-process/src/services/tenantService.ts

```diff
@@ -417,6 +417,60 @@ export function tenantServiceBuilder(
       );
       return updatedTenant;
     },
+    async revokeDeclaredAttribute(
+      {
+        attributeId,
+        organizationId,
+        correlationId,
+      }: {
+        attributeId: AttributeId;
+        organizationId: TenantId;
+        correlationId: string;
+      },
+      logger: Logger
+    ): Promise<Tenant> {
+      logger.info(
+        `Revoking declared attribute ${attributeId} to tenant ${organizationId}`
+      );
+      const requesterTenant = await retrieveTenant(
+        organizationId,
+        readModelService
+      );
+
+      const maybeDeclaredTenantAttribute = requesterTenant.data.attributes.find(
+        (attr): attr is DeclaredTenantAttribute =>
+          attr.id === attributeId && attr.type === "PersistentDeclaredAttribute"
```

Comment:
the code does not take into considerations that the attribute could be already revoked.

this is the current behaviour, it's ok IMO to leave it like this, but we may open a task to make this fail.

cc @beetlecrunch (if I remember your point, failure was the intended behaviour in the future)